client/core/core_test.go

```diff
@@ -4373,6 +4382,46 @@ func TestHandleTradeSuspensionMsg(t *testing.T) {
 	}
 }
 
+func orderNoteFeed(tCore *Core, t *testing.T) chan *OrderNote {
+	t.Helper()
+	ch := tCore.NotificationFeed()
+	outChan := make(chan *OrderNote)
+	go func() {
+		for start := time.Now(); time.Since(start) < time.Second; {
+			select {
+			case n := <-ch:
+				ordNote, ok := n.(*OrderNote)
+				if ok {
+					outChan <- ordNote
+					return
+				}
+			case <-tCtx.Done():
+				return
+			case <-time.NewTimer(time.Second).C:
+				t.Fatal("timed out waiting for notifications")
+				return
+			}
+		}
+	}()
+	return outChan
+}
```

Comment:
The goroutine started by `orderNoteFeed` has a similar issue with `Fatal`, which is not allowed to be called from goroutines other than the one running the test function.  See https://golang.org/pkg/testing/#B.FailNow (called by `Fatal`/`Fatalf`)

Also, there's nothing in the tests that makes the goroutine return before the test returns.  I suggest changing the function to:

```go
func orderNoteFeed(tCore *Core) (orderNotes chan *OrderNote, done func()) {
	orderNotes = make(chan *OrderNote, 1)

	ntfnFeed := tCore.NotificationFeed()
	feedDone := make(chan struct{})
	var wg sync.WaitGroup
	wg.Add(1)
	go func() {
		defer wg.Done()
		for {
			select {
			case n := <-ntfnFeed:
				ordNote, ok := n.(*OrderNote)
				if ok {
					orderNotes <- ordNote
					return // just one OrderNote and done
				}
			case <-tCtx.Done():
				return
			case <-feedDone:
				return
			}
		}
	}()

	done = func() {
		close(feedDone) // close first on return
		wg.Wait()
	}
	return orderNotes, done
}
```

And calling it like:

```go
orderNotes, feedDone := orderNoteFeed(tCore)
defer feedDone()
```

Reply:
That makes sense. Changes made.